ComponentKit/Core/CKChildComponent.h

```diff
@@ -0,0 +1,24 @@
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+
+#import <ComponentKit/CKComponent.h>
+#import <ComponentKit/CKComponentTreeOwner.h>
+
+@interface CKChildComponent : CKComponent <CKComponentTreeOwner>
+
+/**
+ Returns a child component that needs to be rendered from this component.
+
+ @param state The current state of the component.
+ */
+- (CKComponent *)render:(id)state;
```

Comment:
Yes that's the way Litho works. The first version did return a vector of children. However, changing to this approach allowed us to optimize the component tree creation time and memory. 
In most of the cases we actually return a single child anyway. In case we need to return more than one there are two options:
1. Use `CKFlexboxComponent` (Row or Column in Litho)
2. Inherit from `CKGrouopComponent` and override the `renderGroup` method.

In the 2nd case, we have to provide a custom layout.